### PR TITLE
Fix block header calculation

### PR DESCRIPTION
**Pull request by oliviera9**
Block https://algoexplorer.io/block/21139432 contains no transactions. The transaction root hash is `AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=`: its decoding is a vector of 32 zeros. The all-zero vector is not to be encoded when calculating the block hash.

### CODE REVIEW

File: src/algorand_hash.rs [hunk 1]
```diff
@@ -10,6 +10,7 @@ use crate::{
     algorand_errors::AlgorandError,
     algorand_genesis_id::AlgorandGenesisId,
     algorand_types::{Byte, Bytes, Result},
+    crypto_utils::base32_encode_with_no_padding,
```
Comment (oliviera9) [inline src/algorand_hash.rs]: Clippy is complaining about this. Maybe we should remove `#[cfg(test)]` from the two functions you added.

Comment (gskapka) [reply src/algorand_hash.rs]: No need - just remove the import from there and add it on it's own line:

```rust
#[cfg(test)]
use crate::crypto_utils::base32_encode_with_no_padding;
```

File: src/algorand_hash.rs [hunk 2]
```diff
@@ -43,6 +44,15 @@ impl AlgorandHash {
         }
     }
 
+    /// ## Check if all hash elements are 0
+    ///
+    /// This will return true in case all 32 bytes are 0, false otherwise.
+    /// Base64 representation of the all-zero-elements hash is
+    /// AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=
+    pub fn is_zero(self) -> bool {
+        !self.0.iter().any(|el| *el != 0)
```
Comment (gskapka) [inline src/algorand_hash.rs]: No need to iterate over anything, just do:

```suggestion
        self == Self::default()
```

File: src/algorand_hash.rs [hunk 3]
```diff
@@ -191,4 +206,20 @@ mod tests {
         let result = AlgorandHash::from_base_64(&base_64_encoded).unwrap();
         assert_eq!(result, hash);
     }
+
+    #[test]
+    fn should_return_true_if_hash_is_all_zeros() {
+        let hash =
+            AlgorandHash::from_base_64("AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=").unwrap();
```
Comment (gskapka) [inline src/algorand_hash.rs]: Same as above:

```suggestion
            AlgorandHash::default();
```

**Review by gskapka — CHANGES_REQUESTED** [review]
Just a couple of minor things, plus the clippy lints need cleaning up.

Once done, it'll be ready to merge.

**Review by gskapka — APPROVED** [review]
LGTM!

**Pull request merged**